### DISCUSSION TITLE
Expose window events in Native mode

### DISCUSSION THREAD

**Discussion by gavinpotter** [zauberzeug/nicegui#2277] [Ideas / Feature Requests]
"In native mode the app.native.main_window object allows you to access the underlying window. It is an async version of [Window from pywebview](https://pywebview.flowrl.com/guide/api.html#window-object)".

Would it be possible to expose the events associated with the pywebview window. My use  case is that I want to be able to take various actions when the  native window is minimised and restored.

Thanks

Gavin

**Comment by falkoschindler** [comment]
Thanks for the suggestion, @gavinpotter!

I think it would be possible. But it's not trivial:
- We somehow need to get the event registration from the main process to the webview process. A callback can't be pickled, so we need to manage all registered callbacks on the main process and call them when the event happens.
- We need to get the event back to the main process when it happens. We already have a "message queue" and a "response queue". Maybe we need another "event queue".
- I'm not sure if we should only send events that have registered callbacks (less traffic), or simply _every_ possible event (easier to implement).

We could certainly use some help implementing this feature. Would you - or anyone else - like to create a pull request? 🙂

**Comment by gavinpotter** [comment]
That does sound complicated - beyond what I could manage.  I'll look for an alternative.  The approach I'm going to take is to see what else changes when each event occurs - and act on that.  So if, for instance, you maximise a minimised window, the size of the window increases etc which you can capture.
